internal/controller/humiocluster_controller.go

```diff
@@ -196,6 +198,10 @@ func (r *HumioClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		r.ensureHumioClusterKeystoreSecret,
 		r.ensureNoIngressesIfIngressNotEnabled, // TODO: cleanupUnusedResources seems like a better place for this
 		r.ensureIngress,
+		// Propagate TLS settings to pdf-render-service
+		func(ctx context.Context, hc *humiov1alpha1.HumioCluster) error {
+			return r.ensurePdfRenderServiceTLS(ctx, hc)
+		},
```

Comment:
I think the function name and comment here is misleading. It doesn't _just_ propagate TLS configs to the render service object, but straight up also creates it if not already present.

Also, you don't need to do all that `func ...` stuff here. each function listed here is already assumed to have `ctx, hc` as inputs, so you just need to list the function name and the loop will call it:
```suggestion
		r.ensurePdfRenderServiceTLS(ctx, hc)
```